src/netif.c

```diff
@@ -1995,14 +2071,29 @@ static inline int netif_deliver_mbuf(struct rte_mbuf *mbuf,
     struct pkt_type *pt;
     int err;
     uint16_t data_off;
+    struct rte_mbuf *mbuf_cloned = NULL;
 
     assert(mbuf->port <= NETIF_MAX_PORTS);
     assert(dev != NULL);
 
     pt = pkt_type_get(eth_type, dev);
-    if (NULL == pt) {
-        kni_ingress(mbuf, dev, qconf);
-        return EDPVS_OK;
+
+    /*
+    In NETIF_PORT_FLAG_FORWARD2KNI mode.
```

Comment:
`netif_deliver_skb` is too late ? you can add the code block into `lcore_job_recv_fwd` ? so that we can capture VLAN packets.